src/components/GridLayout/GridLayout.js

```diff
@@ -70,27 +69,50 @@ export default class GridLayout extends React.PureComponent {
         return getItemsMeta(this.pluginsRefs);
     };
 
-    reloadItems() {
-        const {
-            editMode,
-            settings: {autoupdateInterval, silentLoading} = {},
-            reloadItems,
-        } = this.context;
-        const {isPageHidden} = this.state;
+    getAutoupdateIntervalMs() {
+        const {settings: {autoupdateInterval} = {}} = this.context;
         const autoupdateIntervalMs = Number(autoupdateInterval) * 1000;
+        return autoupdateIntervalMs;
+    }
+
+    startReloadItems() {
+        const {editMode, settings: {silentLoading} = {}, reloadItems} = this.context;
+        const {isPageHidden} = this.state;
+
+        if (!isPageHidden && !editMode) {
+            this._lastReloadAt = new Date().getTime();
+            return reloadItems(this.pluginsRefs, {silentLoading, noVeil: true});
+        }
+
+        return Promise.resolve();
+    }
+
+    startReloadItemsTimeout(timeoutMs) {
+        this._timeout = setTimeout(() => this.forceReloadItems(), timeoutMs);
+    }
+
+    forceReloadItems() {
+        clearTimeout(this._timeout);
+        this._timeout = undefined;
+        this.reloadItems();
+    }
+
+    reloadItems() {
+        if (this._timeout) return;
+
+        const {settings: {realtimeMode, realtimeModeDelayMs = 5000} = {}} = this.context;
+
+        if (realtimeMode) {
+            this.startReloadItems().then(() => {
+                this.startReloadItemsTimeout(realtimeModeDelayMs);
+            });
+            return;
+        }
+
+        const autoupdateIntervalMs = this.getAutoupdateIntervalMs();
         if (autoupdateIntervalMs) {
-            const timeSinceLastReload = new Date().getTime() - (this._lastReloadAt || 0);
-            const reloadIntervalRemains = autoupdateIntervalMs - timeSinceLastReload;
-
-            if (!isPageHidden && !editMode && reloadIntervalRemains <= 0) {
```

Comment:
Here was a logic for autoupdate for reloading immediately if dash was not in focus for time more that timer setting and later become in focus. I can't see this in PR new changes

Reply:
Thank you for the highlight. It's not an obvious feature.
For the implementation of this feature, I used the `visibilitychange` event.

Reply:
Now there is bug that if we return earlier that interval ended we update it, but we need to wait for rest of time